core/include/traccc/fitting/fitting_algorithm.hpp

```diff
@@ -1,88 +0,0 @@
-/** TRACCC library, part of the ACTS project (R&D line)
- *
- * (c) 2022-2023 CERN for the benefit of the ACTS project
- *
- * Mozilla Public License Version 2.0
- */
-
-#pragma once
-
-// Project include(s).
-#include "traccc/definitions/qualifiers.hpp"
-#include "traccc/edm/track_candidate.hpp"
-#include "traccc/edm/track_state.hpp"
-#include "traccc/fitting/fitting_config.hpp"
-#include "traccc/fitting/kalman_filter/kalman_fitter.hpp"
-#include "traccc/utils/algorithm.hpp"
-
-namespace traccc {
-
-/// Fitting algorithm for a set of tracks
-template <typename fitter_t>
-class fitting_algorithm
```

Comment:
Hmm, why are you changing the name? The motivation of this name and template parameter is using various fitters with the same interface

Reply:
My overall points with these PRs is that we can't have such an interface for our GPU algorithms. Users cannot freely choose between different template specialisations. We'll provide them with a few specific ones, and they'll have to use one of those.

So the algorithm that I'm introducing, `traccc::host::kf_algorithm`, is specifically an algorithm applying a Kálmán fit to the tracks. With one specific setup.

Once/if we have a different valid setup for fitting as well, we'll create another algorithm class. One that will reflect in its name that it is performing a different kind of fit.

In the `src/` directory we can still have a single `traccc::host::details::fit_tracks` function. Which would get a different fitter template argument in all of these cases. But it would not be up to the user to construct the fitter themselves. 🤔

This latter point in my mind will actually make it a lot easier to use these algorithms. Since right now you really need expert level knowledge to just know how to even instantiate one of these algorithms such that they would make any sense.

Reply:
If so, we might need to put those files in `fitting/kalman_filter`. In case we add the triplet fitter, global chi square fitter, DAF, etc, we may not want to dump them and their specialization into the same `fitting/` directory

Reply:
Splitting them would make more sense as far as I can tell.. OK